### ISSUE TITLE
Cockatrice can't handle new card database fields. (Missing tokens with new token file on old cockatrice)

### ISSUE THREAD

**Issue by Jorenthar** [Cockatrice/Cockatrice#1810]
For some reason when I tried to get out a token into play I checked my tokens list, but nothing was in it save for an 4/4 white Angel token. I tried to re-download all card sets and tokens and restarting the client, but nothing worked. Help please!

**Comment by ZeldaZach** [comment]
Did you update your client? What client version are you on? (About -> Cockatrice)

**Comment by Jorenthar** [comment]
Version d55e44e

**Comment by vidartf** [comment]
I also had this issue. Was originally on d55e44e, then installed 323fc06, same issue. Switching the URL manually / resetting fixed it. I think maybe something was wrong with the old tokens URL?

**Comment by Skaeg** [comment]
I was also experiencing this issue on d55e44e. Updating to 0b8f52e fixed the issue.

**Comment by tooomm** [comment]
@Jorenthar @vidartf @Skaeg 
Speaking of the not working version... Was Oracle downloading tokens without problems?
Or did some kind of error happen during token downloading?

Do you happen to have a not working version of `tokens.xml` that you could upload?

In case we don't want to look further and consider this solved with recent versions --> close?

**Comment by Skaeg** [comment]
@tooomm Yes, it was downloading the token.xml file without issue and would install correctly, but when opening Cockatrice, it would blow away most of the entries. I did some searching and found a copy of the corrupted xml file.
[tokens.zip](https://github.com/Cockatrice/Cockatrice/files/165564/tokens.zip)

**Comment by ctrlaltca** [comment]
As reported by @Jorenthar, the file contains a single "Angel" token card:
```
<?xml version="1.0" encoding="UTF-8"?>
<cockatrice_carddatabase version="3">
    <cards>
        <card>
            <name>Angel</name>
            ...
            <color>W</color>
            <manacost></manacost>
            <cmc></cmc>
            <type>Token Creature — Angel</type>
            <pt>4/4</pt>
            <tablerow>2</tablerow>
            <text>Flying</text>
            <token>1</token>
        </card>
    </cards>
</cockatrice_carddatabase>
```

When the last version of cockatrice has been released, the new reverse-related field has been addd to the default token file downloads by oracle. If you're still running an older version of cockatrice, that doesn't support the reverse-related functionality, oracle is probably not able to parse the token file and produces this.

**Comment by Daenyth** [comment]
I thought oracle just downloaded the token file, and it didn't parse it.

Is this really how oracle behaves? Adding a new field shouldn't affect older clients. That's a big mistake on our part if updating that broke everyone who wasn't up to date

**Comment by ctrlaltca** [comment]
> Is this really how oracle behaves? Adding a new field shouldn't affect older clients.

The card database's xml parser actual behavior is to error out if an unknown field is found; i agree this should be changed.

> That's a big mistake on our part if updating that broke everyone who wasn't up to date

We should probably have incremented the cockatrice database version number and used a different url for the new token file.

**Comment by tooomm** [comment]
Could Cockatrice just skip unknown fields? (And print a info msg to some debug logs maybe?)
Introducing and maintaining of several token versions + updating links in Oracle accordingly etc. is not what we really want I guess...

**Comment by tooomm** [comment]
this should be fixed with #1901?
